python/ray/air/util/object_extensions/arrow.py

```diff
@@ -0,0 +1,91 @@
+import pickle
+import typing
+
+import numpy as np
+import pyarrow as pa
+from packaging.version import parse as parse_version
+
+import ray.air.util.object_extensions.pandas
+from ray._private.serialization import pickle_dumps
+from ray._private.utils import _get_pyarrow_version
+from ray.util.annotations import PublicAPI
+
+MIN_PYARROW_VERSION_SCALAR_SUBCLASS = parse_version("9.0.0")
```

Comment:
What happens if we use a lower version of PyArrow? Which APIs do we need?

Reply:
Subclassing extension scalars is only available in PyArrow 9.0. This is needed for the code:
```
@PublicAPI(stability="alpha")
class ArrowPythonObjectScalar(pa.ExtensionScalar):
    def as_py(self) -> typing.Any:
        if not isinstance(self.value, pa.LargeBinaryScalar):
            raise RuntimeError(
                f"{type(self.value)} is not the expected LargeBinaryScalar"
            )
        return pickle.load(pa.BufferReader(self.value.as_buffer()))
```
that allows all of the as_py conversions in the array.

Reply:
@bveeramani WDYT; OK (if so please resolve this conflict).

Reply:
This should be fine.  We'll consider bumping the minimum PyArrow version later.